os/os.go

```diff
@@ -33,6 +33,7 @@ type File interface {
 }
 
 type FS interface {
+	Append(name string) (File, error)
```

Comment:
Would you mind switching this to be a more general `OpenFile` instead? My aim with this interface is to match the Go `os` and/or `io/fs` package functions.

So specifically the signature should be:

```
OpenFile(name string, flag int, perm FileMode) (File, error)
```

Supporting `OpenFile` with all the different flags on `s3fs` may be a project onto itself, so I'd suggest just having that implementation return a "not implemented" error for now. (I don't want to make the job too hard :-) 

The PathSeparator and PathListSeparator additions are good as-is.

Reply:
Done